src/test/java/bio/terra/service/filedata/azure/tables/TableFileDaoTest.java

```diff
@@ -1,105 +1,119 @@
-    assertTrue("Existing row is deleted", exists);
-    boolean result = dao.deleteFileMetadata(tableServiceClient, DATASET_ID, "nonexistentFile");
-    assertFalse("Non-existent row is not deleted", result);
-  }
+  @Nested
+  class TestRetrieveDeleteFileMetadata {
 
-  @Test
-  public void testBatchRetrieveFileMetadata() {
-    FireStoreDirectoryEntry fsDirectoryEntry = new FireStoreDirectoryEntry().fileId(FILE_ID);
-    List<FireStoreDirectoryEntry> directoryEntries = List.of(fsDirectoryEntry);
-    List<FireStoreFile> expectedFiles = List.of(FireStoreFile.fromTableEntity(entity));
-    List<FireStoreFile> files =
-        dao.batchRetrieveFileMetadata(tableServiceClient, DATASET_ID, directoryEntries);
-    assertEquals(
-        "A file record is found for each directory entry", files.size(), expectedFiles.size());
-    assertEquals("The same object is returned", files.get(0), expectedFiles.get(0));
+    @Mock private TableServiceClient tableServiceClient;
+    @Mock private TableClient tableClient;
+    @Autowired private TableFileDao dao;
+
+    @Mock
+    @Qualifier("performanceThreadpool")
+    private ExecutorService executor;
+
+    private final TableEntity entity = createTableEntity(true);
+    private static final String DATASET_ID = UUID.randomUUID().toString();
+
+    @BeforeEach
+    void setUp() {
+      when(tableServiceClient.getTableClient(any())).thenReturn(tableClient);
+      when(tableClient.getEntity(PARTITION_KEY, FILE_ID)).thenReturn(entity);
+    }
+
+    @Test
+    void testRetrieveFileMetadata() {
+      FireStoreFile fileMetadata =
+          dao.retrieveFileMetadata(tableServiceClient, DATASET_ID, FILE_ID);
+      FireStoreFile expected = FireStoreFile.fromTableEntity(entity);
+      assertEquals(fileMetadata, expected, "The same object is returned");
+    }
+
+    @Test
+    void testDeleteFileMetadata() {
+      boolean exists = dao.deleteFileMetadata(tableServiceClient, DATASET_ID, FILE_ID);
+      assertTrue(exists, "Existing row is deleted");
+
+      when(tableClient.getEntity(PARTITION_KEY, "nonexistentFile"))
+          .thenThrow(TableServiceException.class);
+      boolean result = dao.deleteFileMetadata(tableServiceClient, DATASET_ID, "nonexistentFile");
+      assertFalse(result, "Non-existent row is not deleted");
+    }
+
+    @Test
+    void testBatchRetrieveFileMetadata() {
+      FireStoreDirectoryEntry fsDirectoryEntry = new FireStoreDirectoryEntry().fileId(FILE_ID);
+      List<FireStoreDirectoryEntry> directoryEntries = List.of(fsDirectoryEntry);
+      List<FireStoreFile> expectedFiles = List.of(FireStoreFile.fromTableEntity(entity));
+      List<FireStoreFile> files =
+          dao.batchRetrieveFileMetadata(tableServiceClient, DATASET_ID, directoryEntries);
+      assertEquals(
+          files.size(), expectedFiles.size(), "A file record is found for each directory entry");
+      assertEquals(files.get(0), expectedFiles.get(0), "The same object is returned");
```

Comment:
I would use `assertThat()` here instead to check both of these things at once:
```suggestion
      assertThat(files, is(expectedFiles));
```